internal/cluster/host_spec_subresource.go

```diff
@@ -0,0 +1,153 @@
+/*
+ *  Copyright 2023 VMware, Inc.
+ *    SPDX-License-Identifier: MPL-2.0
+ */
+
+package cluster
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/terraform-provider-vcf/internal/network"
+	validation_utils "github.com/vmware/terraform-provider-vcf/internal/validation"
+	"github.com/vmware/vcf-sdk-go/models"
+)
+
+// HostSpecSchema this helper function extracts the Host
+// schema, so that it's made available for both Domain and Cluster creation.
+func HostSpecSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "ID of a vSphere host in the free pool",
+			},
+			"host_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Host name of the vSphere host",
```

Comment:
This concerns me. 

A hostname is generally a short name. For example, if the fully qualified domain name (FQDN) is `sfo-m01-esx01.sfo.rainpole.io` the hostname is usually the first part - `sfo-m01-esx01`.

```suggestion
				Description:  "Fully qualified domain name of the ESXi host",
```
Should this be `fqdn`?



```suggestion
				Description:  "Host name of the vSphere host",
```

Reply:
The API object HostSpec that this subresource is specifically referring to "host_name".

Reply:
https://developer.vmware.com/apis/vcf/4.5.0/data-structures/HostSpec/